doc/release-notes.md

```diff
@@ -246,6 +246,11 @@ in the Low-level Changes section below.
 
 - See the [Mining](#mining) section for changes to `getblocktemplate`.
 
+- The `getrawtransaction` RPC no longer checks the unspent UTXO set for
+  a transaction by default. It continues to check the mempool, a specific
+  block, or the mempool and tx index based on if a blockhash is passed in
```

Comment:
This sentence is a bit imprecise. I think it'd be better to give the exact logic (if called with blockhash: that block is checked; if called without blockhash and txindex disabled: mempool is checked; if called without blockhash and txindex enabled: mempool and txindex is checked)